src/audio/eq_fir.c

```diff
@@ -247,11 +247,17 @@ static struct comp_dev *eq_fir_new(struct sof_ipc_comp *comp)
 	struct comp_data *cd;
 	struct sof_ipc_comp_eq_fir *ipc_fir
 		= (struct sof_ipc_comp_eq_fir *)comp;
-	size_t bs;
+	size_t bs = ipc_fir->size;
 	int i;
 
 	trace_eq("new");
 
+	/* Check first before proceeding with dev and cd that coefficients
+	 * blob size is sane.
+	 */
+	if (bs > SOF_EQ_FIR_MAX_SIZE)
```

Comment:
we probably want to trace this error to report this.

Reply:
yep